app/models/miq_server/worker_management/monitor/validation.rb

```diff
@@ -4,22 +4,50 @@ module MiqServer::WorkerManagement::Monitor::Validation
   def validate_worker(w)
     return true if MiqEnvironment::Command.is_podified?
 
-    time_threshold   = get_time_threshold(w)
-    memory_threshold = get_memory_threshold(w)
+    if !worker_heartbeat_valid?(w)
+      stop_worker(w, MiqServer::NOT_RESPONDING)
+      return false
+    end
 
-    w.validate_active_messages
+    if !worker_memory_valid?(w)
+      stop_worker(w)
+      return false
+    end
 
-    validate_heartbeat(w)
+    return true unless worker_get_monitor_status(w.pid).nil?
```

Comment:
`return true if worker_get_monitor_status(w.pid)`?  The `unless .nil?` is confusing

Reply:
This is also just a move [ref](https://github.com/ManageIQ/manageiq/pull/19609/files#diff-57a2568ae37208b39c60baa82b4c3a9cL22). I was hoping stuff like that would be obvious in the commit by commit view.

Reply:
I saw the move, but I was questioning not changing it in a follow up

Reply:
Okay, yeah, I can fix this one. I guess there's no chance we put `false` in the status and don't want this to return true in that case :laughing:

Reply:
Wait ... this call isn't actually changing the behavior ... I could just remove this line I think, right?

Reply:
Ohh this should be between the heartbeat and memory checks

I guess we're only checking for memory violation if we have a monitor status? Weird ...